atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/persistentlock/LockStore.java

```diff
@@ -101,6 +104,21 @@ public void releaseLock(LockEntry lockEntry) throws CheckAndSetException {
         log.info("Successfully released the persistent lock: {}", lockEntry);
     }
 
+    public LockEntry getLockEntryWithLockId(PersistentLockId lockId) {
+        List<LockEntry> locksWithId = allLockEntries().stream()
+                .filter(lockEntry -> Objects.equals(lockEntry.instanceId(), lockId.value()))
+                .collect(Collectors.toList());
+
+        Preconditions.checkArgument(
+                locksWithId.size() == 1,
+                "Expected a single lock with id %s but found %s matching locks [%s]",
+                lockId,
+                locksWithId.size(),
+                locksWithId);
+
+        return Iterables.getOnlyElement(locksWithId);
```

Comment:
note: Iterables actually throws IllegalArgumentException if the size==1 is untrue (though it doesn't give you the size and only prints out the first 4 elements).